src/ServicePulse.Host/vue/src/composables/serviceMonitoringEndpoints.ts

```diff
@@ -0,0 +1,163 @@
+import { ref } from "vue";
+import { useFetchFromMonitoring, useIsMonitoringDisabled } from "./serviceServiceControlUrls";
+import { monitoringConnectionState } from "./serviceServiceControl";
+import { useGetExceptionGroups } from "./serviceMessageGroup";
+import { type Endpoint, type GroupedEndpoint, type EndpointGroup, type EndpointDetails, emptyEndpointMetrics } from "@/resources/Endpoint";
+
+/**
+ * @returns the max number of segments in a array of endpoint object names
+ */
+export function useFindEndpointSegments(endpoints: Endpoint[]) {
+  return endpoints.reduce(function (acc, cur) {
+    return Math.max(acc, cur.name.split(".").length - 1);
+  }, 0);
+}
+
+/**
+ * @param historyPeriod - The history period value.  The default is (1)
+ * @returns A array of monitoring endpoint objects
+ */
+export async function useGetAllMonitoredEndpoints(historyPeriod = 1) {
+  let endpoints: Endpoint[] = [];
+  if (!useIsMonitoringDisabled() && !monitoringConnectionState.unableToConnect) {
+    try {
+      const response = await useFetchFromMonitoring(`${`monitored-endpoints`}?history=${historyPeriod}`);
+      const data = response && (await response.json());
+      endpoints = data ?? [];
+      await addEndpointsFromScSubscription(endpoints);
+    } catch (error) {
+      console.error(error);
```

Comment:
Consider not handling error here

Reply:
the general flow is that an error here shouldn't cause the site to fall over, just blank out the endpoint list until connection is re-established.